pkgs/applications/blockchains/solana-validator/default.nix

```diff
@@ -56,8 +60,15 @@ rustPlatform.buildRustPackage rec {
     inherit hash;
   };
 
-  # partly inspired by https://github.com/obsidiansystems/solana-bridges/blob/develop/default.nix#L29
-  inherit cargoHash;
+  patches = [ ./rustc-1.7x.patch ];
```

Comment:
upstream patches using [fetchpatch](https://nixos.org/manual/nixpkgs/stable/#fetchpatch) are preferred, looks like this change is contained in the following commit, https://github.com/solana-labs/solana/commit/9e703f85de4184f577f22a1c72a0d33612f2feb1

```nix
(fetchpatch {
  url = "https://github.com/solana-labs/solana/commit/9e703f85de4184f577f22a1c72a0d33612f2feb1.patch";
  hash = "";
  includes = [ "path/to/account_info.rs" ];
})
```

Reply:
what do I do if it doesnt apply cleanly?

Reply:
```nix
  patches = [
    (fetchpatch {
       url = "https://github.com/solana-labs/solana/commit/9e703f85de4184f577f22a1c72a0d33612f2feb1.patch";
       hash = "sha256-bAKTIQ6FhTk6bIddYULwLfdH5kzNPw1ltXJEfawtAXg=";
       includes = [ "sdk/program/src/account_info.rs" ];
     })
  ];
```
applies cleanly for me.

Reply:
if fetchpatch seems to be doing the wrong thing, it might be because the hash is stale. if you modify includes, you also need to reset the hash otherwise you'll get stale data.

Reply:
Yeah, sorry I missed the `includes` part